### PR TITLE
fix(errorReporting): fix error with .json() on a 204

### LINKED ISSUE #25: Console error after calling error reporting URL
# 🐞 Bug Report

## Describe the bug

After an error that results in calling the reporting URL, this console error is printed:
```
Uncaught (in promise) SyntaxError: Unexpected end of JSON input
```
It appears to be due to calling `.toJson` on the result of the call, but the default one-app reporting URL returns a 204 response with no body.

## To Reproduce

1. Trigger a reporting error.
2. Check the log.

## Expected behavior

No console error.

## System information

- OS: Mac
- Browser (if applies): Chrome
- Version of one-app-ducks: 4.1.1
- Node version: 12.8.1

### PR DESCRIPTION
**Pull request by JamesSingleton**
## Description

Fixes #25

## How Has This Been Tested?

Unit tests and tested locally with One App.

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] My changes are in sync with the code style of this project.
- [ ] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] This change impacts caching for client browsers.
- [ ] This change adds additional environment variable requirements for one-app-ducks users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

**Review by smackfu — APPROVED** [review]

File: src/errorReporting.js [hunk 1]
```diff
@@ -126,7 +126,11 @@ function thenSendErrorReport({
           },
         }).then((response) => {
           if (response.ok) {
-            return response.json();
+            const textResponse = response.clone();
+            const jsonResponse = response.clone();
+            return jsonResponse.json()
+              .catch(() => textResponse.text())
+              .then((data) => data);
```
Comment (JAdshead) [inline src/errorReporting.js:133]: whats the extra `.then` for ?

Comment (JAdshead) [inline src/errorReporting.js:133]: any reason for the extra `.then`
```suggestion
              .catch(() => textResponse.text());
```

Comment (10xLaCroixDrinker) [reply src/errorReporting.js:133]: also wondering if this has a purpose

Comment (JamesSingleton) [reply src/errorReporting.js]: Resolved in c06ac75

**Review by infoxicator — APPROVED** [review]

**Review by JAdshead — APPROVED** [review]

**Review by Francois-Esquire — APPROVED** [review]

File: __tests__/errorReporting.spec.js [hunk 2]
```diff
@@ -396,6 +396,46 @@ describe('error reporting', () => {
...
+          msg: testError.message,
+          stack: testError.stack,
+          href: 'about:blank',
+          otherData,
+        },
+      ];
+
+      return store.dispatch(addErrorToReport(testError, otherData))
+        .then(() => {
+          expect(fetch).not.toThrow();
```
Comment (Francois-Esquire) [inline __tests__/errorReporting.spec.js:428]: Wondering if this line would yield the expected behavior. Normally a callback `expect(() => fetch()).not.toThrow()` would be a safer bet. Also, being that `fetch` is asynchronous, are we capturing the error if it were to throw (or reject) for fetch?

**Pull request merged**